app/graphql/utilities/searchers.rb

```diff
@@ -4,4 +4,11 @@ def search(kwargs)
       Event.where(**kwargs)
     end
   end
+
+  class Bucket
+    def search(kwargs)
+      time_bucket = kwargs.delete(:time_bucket)
+      Event.select("time_bucket('#{time_bucket}', event_time) AS bucket, count(*)").group("bucket").where(**kwargs)
```

Comment:
I think that they are validated at the GraphQL stage. Since all of them have to be fields and are all set null: false with a set type.

Although we could set those to optional if we think people will want that.